### PR TITLE
Iteration4 al

### PR DESCRIPTION
**Pull request by alex-latham**
Add winningest_coach, worst_coach methods to SeasonStats, StatTracker, and associated test files. Also moves a method into Stats which pulls all game_teams rows for a given season.

### CODE REVIEW

File: test/stat_tracker_test.rb [hunk 1]
```diff
@@ -115,4 +115,21 @@ def test_it_can_best_fans
   def test_it_can_worst_fans
     assert_equal ["Real Salt Lake", "Minnesota United FC"], @stat_tracker.worst_fans
   end
+
+  def test_game_id_tracks_with_season
+    # this test is to prove that every game id tracks exactly with
+    # its season. E.g., game ids beginning with 2013 always have a
+    # season that begins with 2013.
+    @stat_tracker.games.each do |game|
+      assert_equal game.game_id[0..3], game.season[0..3]
```
Comment (whitneykidd) [inline test/stat_tracker_test.rb:124]: Oh my god she's beautiful 😻

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -21,8 +21,8 @@ def initialize(locations)
     @game_teams = csv_data(locations[:game_teams], GameTeam)
     @game_stats = GameStats.new(@games)
     @league_stats = LeagueStats.new(@games, @teams, @game_teams)
-    @season_stats = SeasonStats.new(@games, @game_teams, @teams)
-    @team_stats = TeamStats.new(@games, @game_teams, @teams)
+    @season_stats = SeasonStats.new(@games, @teams, @game_teams)
+    @team_stats = TeamStats.new(@games, @teams, @game_teams)
```
Comment (Maxwell-Baird) [inline lib/stat_tracker.rb:25]: good job catching the variable name placement

File: lib/stats.rb [hunk 3]
```diff
@@ -1,8 +1,15 @@
 class Stats
   attr_reader :games, :teams, :game_teams
+
   def initialize(games, teams, game_teams)
     @games = games
     @teams = teams
     @game_teams = game_teams
   end
+
+  def season_game_teams(season_param)
```
Comment (tylertomlinson) [inline lib/stats.rb:10]: Good Work!!!

Comment (tylertomlinson) [reply lib/stats.rb:10]: We can mock the test later on

File: test/stats_test.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    assert_instance_of Stats, @stats
+  end
+
+  def test_it_has_attributes
+    assert_equal @games, @stats.games
+    assert_equal @teams, @stats.teams
+    assert_equal @game_teams, @stats.game_teams
+  end
+
+  def test_it_can_return_all_games_in_a_season
```
Comment (tylertomlinson) [inline test/stats_test.rb:24]: mock

**Pull request merged**